pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-standalone/src/main/java/org/apache/pinot/plugin/ingestion/batch/standalone/SegmentGenerationJobRunner.java

```diff
@@ -222,6 +229,15 @@ public void run()
         }
       }
       _segmentCreationTaskCountDownLatch.await();
+
+      if (!_failures.isEmpty()) {
+        StringBuilder msg = new StringBuilder("Failed to generate Pinot segment for file(s) - ");
+        _failures.forEach(p -> {
+          msg.append(msg.length() > 0 ? ", " : "");
```

Comment:
This will always append `", "` because `msg.length()` will never be 0. If we change `_failures` to `Queue<String>`, we can use `StringUtils.join(_failures, ", ")`

Reply:
This code no longer exists, after I changed things to have a single exception (fail fast) as per our discussion.